apps/mobile/src/features/HistoryTransactionDetails/HistoryTransactionDetails.container.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from 'react'
+import { ScrollView, View } from 'tamagui'
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native'
+import { Stack } from 'expo-router'
+
+import { LoadingTx } from '@/src/features/ConfirmTx/components/LoadingTx'
+import { Alert } from '@/src/components/Alert'
+import { useTransactionsGetTransactionByIdV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { useDefinedActiveSafe } from '@/src/store/hooks/activeSafe'
+import { HistoryTransactionView } from '@/src/features/HistoryTransactionDetails/components/HistoryTransactionView'
+import { HistoryTransactionInfo } from '@/src/features/HistoryTransactionDetails/components/HistoryTransactionInfo'
+import { ViewOnExplorerButton } from '@/src/features/HistoryTransactionDetails/components/ViewOnExplorerButton'
+import { getHeaderTitle } from '@/src/features/HistoryTransactionDetails/utils/header'
+
+function HistoryTransactionDetailsContainer() {
+  const txId = useRoute<RouteProp<{ params: { txId: string } }>>().params.txId
+  const activeSafe = useDefinedActiveSafe()
+  const navigation = useNavigation()
+
+  useEffect(() => {
+    navigation.setOptions({
+      title: 'Transaction details',
+    })
+  }, [navigation])
+
+  const {
+    data: txDetails,
+    isFetching: isLoading,
+    isError,
+  } = useTransactionsGetTransactionByIdV1Query({
+    chainId: activeSafe.chainId,
+    id: txId,
+  })
+
+  if (isLoading || !txDetails) {
+    return <LoadingTx />
+  }
+
+  if (isError) {
+    return (
+      <View margin="$4">
+        <Alert type="error" message="Error fetching transaction details" />
+      </View>
+    )
+  }
```

Comment:
Should the error block be above the loading? In case there is an error I can imagine that txDetails doesn't exist so we would always land in the loading block. Ideally this is covered by a test.

Reply:
good catch!